src/Domain/ShoppingList.php

```diff
@@ -99,12 +99,15 @@ public function isArchived(): bool
 
     /**
      * Set whether an item is archived.
-     *
+     * @param bool $isArchived
      * @return self
      */
-    public function setArchived(): self
+    public function setArchived(bool $isArchived): self
     {
-       return $this;
+        if (!empty($isArchived)) {
```

Comment:
Question: not sure why you've put this in? Should we conditionally be setting the value, or should it just be set to the value that is given to the function (which is already type-hinted as `bool`)?